stl/inc/xhash

```diff
@@ -1841,6 +1841,7 @@ protected:
         }
     }
 
+    template <bool _Multi2 = _Traits::_Multi, enable_if_t<_Multi2, int> = 0>

```

Comment:
why do we need the first bool parameter? Why not `template<enable_if_t<_Traits::_Multi, int> = 0>`?

Reply:
This is necessary to "delay" the evaluation. Because `_Traits::_Multi` is a known value when processing the class definition, the compiler might "eagerly" try to determine what `enable_if_t<KNOWN BOOL VALUE, int>` is (and if it's `false`, the nested `type` won't exist, and we'll get a hard error). By introducing `_Multi2`, the compiler doesn't know for sure what it's going to be (by convention, we always use the default value of `_Traits::Multi`).

This technique is used elsewhere in the STL, e.g.:
https://github.com/microsoft/STL/blob/47cd70315d560e4fe1ac699c0eb8ec8bfc5cb658/stl/inc/xtree#L1228-L1237

With concepts, I think we can just say `requires` instead of introducing this extra layer, but we can't unconditionally use concepts here.

Reply:
ah, thanks

Reply:
If the intent is simply to avoid instantiation when the class is instantiated, `template <int = 0>` would work as well. IIRC we've used `<int = 0>` and/or `<class = void>` in the past as a minimal idiomatic expression of "this is a template for no reason other than that we need it to be a template".

Reply:
Since the function is multi-only, I'd prefer to keep the constraint and drop the redundant `static_assert`, if you feel that that would be an improvement.

Reply:
(Since you've both approved, I'm assuming that everyone is reasonably happy with the current state, so I'm not planning to make further changes that would reset testing.)